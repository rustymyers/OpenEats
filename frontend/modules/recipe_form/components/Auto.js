import React from 'react'
import Autocomplete from 'react-autocomplete';
import {
    injectIntl,
    IntlProvider,
    defineMessages,
    formatMessage
} from 'react-intl';

export var Auto = injectIntl(React.createClass({
  getInitialState: function() {
    return {
      val: this.props.value || '',
      data: this.props.data || '',
    };
  },

  handleChange(event) {
    this.setState({val: event.target.value});
    if(this.props.change) {
      this.props.change(event.target.name, event.target.value);
    }
  },

  handleSelect(value) {
    this.setState({val: value});
    if(this.props.change) {
      this.props.change(this.props.name, value);
    }
  },

  matchStateToTerm (state, value) {
    return (
      state.toLowerCase().indexOf(value.toLowerCase()) !== -1
    )
  },

  sortStates (a, b, value) {
    return (
      a.toLowerCase().indexOf(value.toLowerCase()) >
      b.toLowerCase().indexOf(value.toLowerCase()) ? 1 : -1
    )
  },

  renderItems (items) {
    return items.map((item) => {
      return item
    })
  },

  render: function () {
    const {formatMessage} = this.props.intl;
    const messages = defineMessages({
      loading: {
        id: 'autocomplete.loading',
        description: 'Loading message',
        defaultMessage: 'Loading...',
      },
      no_matches: {
        id: 'autocomplete.no_matches',
        description: 'No matches found message',
        defaultMessage: 'No matches for {value}'
      }
    });

    return (
      <div className={this.props.size} key={this.props.id}>
        {this.props.label ? <label>{this.props.label}</label> : null}
        <Autocomplete
          wrapperProps={{className: "form-group autocomplete", }}
          wrapperStyle={{}}
          value={this.state.val}
          inputProps={{name: this.props.name, className: "form-control", placeholder: this.props.placeholder}}
          items={this.state.data}
          getItemValue={(item) => item}
          shouldItemRender={this.matchStateToTerm}
          sortItems={this.sortStates}
          onChange={this.handleChange}
          onSelect={this.handleSelect}
          renderItem={(item, isHighlighted) => (
            <div
              className={isHighlighted ? 'item highlightedItem' : 'item'}
              key={item}
            >{item}</div>
          )}
          renderMenu={(items, value) => (
            <div className="menu">
                { this.state.loading ? (
                  <div className="item">{ formatMessage(messages.loading) }</div>
                ) : items.length === 0 ? (
                  <div className="item">{ formatMessage(messages.no_matches, {value: value}) }</div>
                ) : this.renderItems(items)}
              </div>
          )}
        />
      </div>
    )
  }
}));
