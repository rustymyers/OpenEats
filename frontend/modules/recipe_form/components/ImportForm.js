import React from 'react'
import {
    injectIntl,
    IntlProvider,
    defineMessages,
    formatMessage
} from 'react-intl';
import ImportActions from '../actions/ImportActions';
import { ImportStore, CHANGE_EVENT } from '../stores/ImportStore';

require("./../css/import.scss");
let Spinner = require('react-spinkit');


class ImportForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = this.getStateFromStore();

    this.ImportRecipe = this.ImportRecipe.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);
    this.getStateFromStore = this.getStateFromStore.bind(this);
    this.toggleSupportedSites = this.toggleSupportedSites.bind(this);
    this._onChange = this._onChange.bind(this);
  }

  _onChange() {
    this.setState(this.getStateFromStore());
  }

  getStateFromStore() {
    return {
      source: ImportStore.getStateValue('source'),
      loading: ImportStore.getStateValue('loading'),
      error: ImportStore.getStateValue('error'),
      websites: ImportStore.getStateValue('websites'),
      supportedSites: ImportStore.getStateValue('supportedSites'),
      showSupportedSite: ImportStore.getStateValue('showSupportedSite'),
    };
  }

  componentDidMount() {
    ImportActions.init();
    ImportStore.addChangeListener(CHANGE_EVENT, this._onChange);
  }

  componentWillUnmount() {
    ImportStore.removeChangeListener(CHANGE_EVENT, this._onChange);
  }

  toggleSupportedSites() {
    ImportActions.toggleSupportedSites();
  }

  handleChange (event) {
    this.setState({source: event.target.value});
  }

  handleKeyDown (event) {
    if (event.which === 13) {
      this.ImportRecipe(event);
    }
  }

  ImportRecipe(e) {
    e.preventDefault();
    ImportActions.importRecipe(this.state.source);
  }

  render() {
    const {formatMessage} = this.props.intl;
    const messages = defineMessages({
      header: {
        id: 'recipe.import.header',
        description: 'Rating source placeholder',
        defaultMessage: 'Import A Recipe',
      },
      label: {
        id: 'recipe.import.source_label',
        description: 'Enter a URL to gather recipe data',
        defaultMessage: 'Enter a URL to gather recipe data',
      },
      websites: {
        id: 'recipe.import.websites',
        description: 'Supported Websites',
        defaultMessage: 'Supported Websites',
      },
      import: {
        id: 'recipe.import.import',
        description: 'Import',
        defaultMessage: 'Import',
      },
      importing: {
        id: 'recipe.import.importing',
        description: 'Importing. Please wait',
        defaultMessage: 'Importing',
      },
      error: {
        id: 'recipe.import.error',
        description: 'Error',
        defaultMessage: 'Error',
      },
      error_message_1: {
        id: 'recipe.import.error_message_1',
        description: 'Unknown Error',
        defaultMessage: 'Something went wrong! Please try again.',
      },
      error_message_2: {
        id: 'recipe.import.error_message_2',
        description: 'Website not supported',
        defaultMessage: 'This Website is not supported. ',
      },
      error_message_3: {
        id: 'recipe.import.error_message_3',
        description: 'Invalid URL',
        defaultMessage: 'Please enter a valid URL.',
      },
      hide_sites: {
        id: 'recipe.import.hide_sites',
        description: 'hide supported sites',
        defaultMessage: 'show supported sites',
      },
      show_sites: {
        id: 'recipe.import.show_sites',
        description: 'show supported sites',
        defaultMessage: 'show supported sites',
      }
    });

    let error_message = '';
    if (this.state.error == 1) {
      error_message = formatMessage(messages.error_message_1)
    } else if (this.state.error == 2) {
      error_message = formatMessage(messages.error_message_2)
    } else if (this.state.error == 3) {
      error_message = formatMessage(messages.error_message_3)
    }

    let websites = this.state.supportedSites.map(function (website) {
      return <a key={ website } href={ "http://" + website } className="list-group-item col-md-6 col-xs-12">{ website }</a>;
    });

    let supportedWebsites = (
      <div className="row list-group">
        <a className="list-group-item disabled col-xs-12">{ formatMessage(messages.websites) }</a>
        { websites }
      </div>
    );

    let toggle = (
      <p>
        { formatMessage(messages.show_sites) }
        <span className="glyphicon glyphicon glyphicon-zoom-in"/>
      </p>
    );
    if (this.state.showSupportedSite) {
      toggle = (
        <p>
        { formatMessage(messages.hide_sites) }
          <span className="glyphicon glyphicon glyphicon-zoom-out"/>
        </p>
      );
    }

    return (
      <div className="container">
        <div className="row import-recipe">
          <div className="col-lg-offset-2 col-lg-8">
            { (this.state.error !== 0) ?
              <div className="alert alert-danger" role="alert">
                <span className="glyphicon glyphicon-exclamation-sign"
                      aria-hidden="true"/>&nbsp;
                <span>{ formatMessage(messages.error) }: </span>
                { error_message }
              </div> : null
            }
            <div className="import-recipe-heading">
              <h3>{ formatMessage(messages.header) }</h3>
              <div className="pull-right" onClick={ this.toggleSupportedSites }>
                { toggle }
              </div>
            </div>
            { (this.state.showSupportedSite) ? supportedWebsites : null }
            <input
              type="text"
              autoFocus="true"
              className="form-control"
              placeholder={ formatMessage(messages.label) }
              value={ this.state.value }
              onChange={ this.handleChange }
              onKeyDown={ this.handleKeyDown }
            />

            { (!this.state.loading) ?
              <button
                type="button"
                className="btn btn-lg btn-primary btn-block"
                onClick={ this.ImportRecipe }
              >{ formatMessage(messages.import) }</button>
              :
              <button type="button" className="btn btn-primary btn-lg btn-block" disabled>
                { formatMessage(messages.importing) }
                <Spinner className="spinner-obj" spinnerName="wave" noFadeIn />
              </button>
            }
          </div>
        </div>
      </div>
    )
  }
};

module.exports.ImportForm = injectIntl(ImportForm);