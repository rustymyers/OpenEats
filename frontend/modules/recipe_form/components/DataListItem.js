import React from 'react'
import {
    injectIntl,
    IntlProvider,
    defineMessages,
    formatMessage
} from 'react-intl'

import { TextArea, Input } from '../../common/form/FormComponents'
import { Auto } from './Auto'
import { measurements } from '../../common/config'

class Direction extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      step: this.props.data ? this.props.data.step : 0,
      title: this.props.data ? this.props.data.title : ''
    };

    this.update = this.update.bind(this);
  }

  update(name, value) {
    this.setState({ [name]: value });
    if(this.props.change) {
      this.props.change(name, value);
    }
  }

  render() {
    const {formatMessage} = this.props.intl;
    const messages = defineMessages({
      step_label: {
        id: 'directions.step_label',
        description: 'Step # label',
        defaultMessage: 'Step {step, number}',
      },
      textarea_placeholder: {
        id: 'directions.textarea_placeholder',
        description: 'Step # label',
        defaultMessage: 'Direction',
      }
    });

    return (
      <div className="direction row">
        <label className="col-sm-1 hidden-xs">
          { formatMessage(messages.step_label, {step: this.state.step}) }:
        </label>
        <label className="col-sm-1 visible-xs">
          { this.state.step }:
        </label>
        <TextArea
          name="title"
          type="text"
          placeholder={ formatMessage(messages.textarea_placeholder) }
          size="col-sm-10 col-xs-9"
          change={ this.update }
          value={ this.state.title }/>
        <div className="col-sm-1 col-xs-2">
          <div className="form-group">
            <button onClick={ this.props.deleteData } className="btn btn-danger glyphicon glyphicon-remove" />
          </div>
        </div>
      </div>
    )
  }
}

class Ingredient extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      title: this.props.data ? this.props.data.title : '',
      quantity: this.props.data ? this.props.data.quantity : 0, 
      measurement: this.props.data ? this.props.data.measurement : ''
    };

    this.update = this.update.bind(this);
  }

  update(name, value) {
    this.setState({ [name]: value });
    if(this.props.change) {
      this.props.change(name, value);
    }
  }

  render() {
    const {formatMessage} = this.props.intl;
    const messages = defineMessages({
      name_placeholder: {
        id: 'ingredients.name_placeholder',
        description: 'Ingredients name placeholder',
        defaultMessage: 'Name',
      },
      quantity_placeholder: {
        id: 'ingredients.quantity_placeholder',
        description: 'Ingredients quantity placeholder',
        defaultMessage: 'Quantity',
      },
      measurement_placeholder: {
        id: 'ingredients.measurement_placeholder',
        description: 'Ingredients measurement placeholder',
        defaultMessage: 'Measurement',
      }
    });

    return (
      <div className="ingredient" key={this.props.id}>
        <Input
          name="quantity"
          type="number"
          placeholder={ formatMessage(messages.quantity_placeholder) }
          size="col-sm-3 col-xs-12"
          change={ this.update }
          value={ this.state.quantity }
        />
        <Auto
          name="measurement"
          data={measurements}
          type="text"
          placeholder={ formatMessage(messages.measurement_placeholder) }
          size="col-sm-4 col-xs-12"
          change={ this.update }
          value={ this.state.measurement }
        />
        <Input
          name="title"
          type="text"
          placeholder={ formatMessage(messages.name_placeholder) }
          size="col-sm-4 col-xs-12"
          change={ this.update }
          value={ this.state.title }
        />
        <div className="col-xs-1">
          <div className="form-group">
            <button onClick={ this.props.deleteData } className="btn btn-danger glyphicon glyphicon-remove" />
          </div>
        </div>
      </div>
    )
  }
}

module.exports.Direction = injectIntl(Direction);
module.exports.Ingredient = injectIntl(Ingredient);