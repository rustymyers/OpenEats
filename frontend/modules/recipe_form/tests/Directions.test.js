import React from 'react';
import { DirectionList } from '../components/DataList';
import createComponentWithIntl from '../../../jest_mocks/createComponentWithIntl';


import data from './data';

test('DirectionList component test', () => {
  const component = createComponentWithIntl(
    <DirectionList data={ data.directions }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Add a new empty direction to the list.
  tree.children[0].children[0].children[0].children[8].props.onClick(
    { preventDefault() {} }
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Then remove it.
  tree.children[0].children[0].children[0].children[8].children[3].children[0].children[0].props.onClick(
    { preventDefault() {} }, 8
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
