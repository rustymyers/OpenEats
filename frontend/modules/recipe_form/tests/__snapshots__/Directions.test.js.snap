// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectionList component test 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-12"
  >
    <div
      className="panel panel-default"
    >
      <div
        className="panel-body"
      >
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 1
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            1
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Brown the ground pork and ground sirlion in a medium pan. Add a teaspoon of sereano pepper while browning the meat. Season with kosher salt and pepper."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 2
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            2
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Chop the onion, bell pepper and one Serrano pepper and place them in a large pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 3
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            3
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Open up and drain both cans of kidney beans and add them to the large pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 4
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            4
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Open up both cans of stewed chili style tomatoes and add them to the pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 5
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            5
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Drain the grease away from the browned meat and add the meat to the pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 6
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            6
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Pour in the tomato juice over the meat mixture."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 7
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            7
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Add kosher salt, black pepper, two table spoons of chili powder, and two teaspoons of ground cumin. Stir well."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 8
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            8
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Cook slowly over medium low heat for an hour. If it starts to bubble turn down the heat. Taste during the cooking process to check the seasoning add more to taste."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <a
          className="btn btn-success add-ing"
          href="#"
          onClick={[Function]}
        >
          Next direction
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`DirectionList component test 2`] = `
<div
  className="row"
>
  <div
    className="col-xs-12"
  >
    <div
      className="panel panel-default"
    >
      <div
        className="panel-body"
      >
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 1
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            1
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Brown the ground pork and ground sirlion in a medium pan. Add a teaspoon of sereano pepper while browning the meat. Season with kosher salt and pepper."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 2
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            2
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Chop the onion, bell pepper and one Serrano pepper and place them in a large pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 3
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            3
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Open up and drain both cans of kidney beans and add them to the large pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 4
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            4
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Open up both cans of stewed chili style tomatoes and add them to the pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 5
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            5
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Drain the grease away from the browned meat and add the meat to the pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 6
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            6
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Pour in the tomato juice over the meat mixture."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 7
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            7
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Add kosher salt, black pepper, two table spoons of chili powder, and two teaspoons of ground cumin. Stir well."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 8
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            8
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Cook slowly over medium low heat for an hour. If it starts to bubble turn down the heat. Taste during the cooking process to check the seasoning add more to taste."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 9
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            9
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <a
          className="btn btn-success add-ing"
          href="#"
          onClick={[Function]}
        >
          Next direction
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`DirectionList component test 3`] = `
<div
  className="row"
>
  <div
    className="col-xs-12"
  >
    <div
      className="panel panel-default"
    >
      <div
        className="panel-body"
      >
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 1
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            1
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Brown the ground pork and ground sirlion in a medium pan. Add a teaspoon of sereano pepper while browning the meat. Season with kosher salt and pepper."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 2
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            2
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Chop the onion, bell pepper and one Serrano pepper and place them in a large pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 3
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            3
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Open up and drain both cans of kidney beans and add them to the large pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 4
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            4
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Open up both cans of stewed chili style tomatoes and add them to the pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 5
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            5
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Drain the grease away from the browned meat and add the meat to the pot."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 6
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            6
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Pour in the tomato juice over the meat mixture."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 7
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            7
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Add kosher salt, black pepper, two table spoons of chili powder, and two teaspoons of ground cumin. Stir well."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <div
          className="direction row"
        >
          <label
            className="col-sm-1 hidden-xs"
          >
            Step 8
            :
          </label>
          <label
            className="col-sm-1 visible-xs"
          >
            8
            :
          </label>
          <div
            className="col-sm-10 col-xs-9"
          >
            <div
              className="form-group null"
            >
              <textarea
                className="form-control"
                name="title"
                onChange={[Function]}
                placeholder="Direction"
                rows={undefined}
                type="text"
                value="Cook slowly over medium low heat for an hour. If it starts to bubble turn down the heat. Taste during the cooking process to check the seasoning add more to taste."
              />
            </div>
          </div>
          <div
            className="col-sm-1 col-xs-2"
          >
            <div
              className="form-group"
            >
              <button
                className="btn btn-danger glyphicon glyphicon-remove"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>
        <a
          className="btn btn-success add-ing"
          href="#"
          onClick={[Function]}
        >
          Next direction
        </a>
      </div>
    </div>
  </div>
</div>
`;
